proposals/dictionary-expressions.md

```diff
@@ -312,11 +228,11 @@ The runtime has committed to supplying these new CollectionBuilder methods that
 
 The elements of a collection expression are evaluated in order, left to right. Each element is evaluated exactly once, and any further references to the elements refer to the results of this initial evaluation.
 
-If the target is a dictionary type, and collection expression's first element is an `expression_element`, and the type of that element is convertible to some `IEqualityComparer<TKey>`, then:
+If the target is a dictionary type, and collection expression's first element is an `expression_element`, and the type of that element is convertible to some [*comparer*](Comparer-support), then:
 
-1. If using a constructor to instantiate the value, the constructor must take a single parameter whose type is convertible to `IEqualityComparer<TKey>`.  The first `element_expression` value will be passed to this parameter.
-2. If using a `create method`, the method must have a parameter whose type is convertible to `IEqualityComparer<TKey>` as one of its parameters. The first `element_expression` value will be passed to this parameter.
-3. If creating an interface, this comparer will be used as the equality comparer controlling the behavior of the final type (synthesized or otherwise) instantiated.
+1. If using a constructor to instantiate the value, the constructor must take a single parameter whose type is some [*comparer*](Comparer-support) type.  The first `element_expression` value will be passed to this parameter.
+2. If using a `create method`, the method must have a parameter whose type is some [*comparer*](Comparer-support) as one of its parameters. The first `element_expression` value will be passed to this parameter.
+3. If creating an interface, this [*comparer*](Comparer-support) must be some `IEqualityComparer<TKey>` type. That comparer will be used to  control the behavior of the final type (synthesized or otherwise).  This means that instantiating interfaces only supports hashing semantics, not ordered semantics. 
```

Comment:
Tagging @stephentoub .  If you guys added an `ISortedDictionary<>` and `IReadOnlySortedDictionary<>` we could key off of those to create sorted versions automatically.  That can also come in the future.